packages/components/src/internal/components/domainproperties/entities/EntityDetailsForm.tsx

```diff
@@ -137,7 +137,7 @@ export class EntityDetailsForm extends React.PureComponent<EntityDetailsProps, a
                             type="text"
                             placeholder={nameExpressionPlaceholder}
                             onChange={onFormChange}
-                            defaultValue={getEntityNameExpressionValue(formValues, data)}
+                            value={getEntityNameExpressionValue(formValues, data)}
```

Comment:
Just for my own edification, I wasn't able to repro the case where react doesn't update the value, but I might be doing the wrong thing. Within a biologics dataclass, and on the server side, I went into a dataclass design and edited the 'Naming Pattern' value. Is this the wrong series of steps to repro the bug?

Reply:
The issue 45128 is a repro of value vs defaultValue. defaultValue doesn't bind to UI, but value does. With defaultValue, the formValues is updated to blank (due to the bug), but the UI still displays 'Source-${genId}'